Sources/Services/SessionController/Session.swift

```diff
@@ -72,21 +72,31 @@ class Session {
         case .started(let startedAt):
             // Capture endedAt now, as opposed to when the timer fires
             let endedAt = Date()
+            
+            var backgroundTaskID: UIBackgroundTaskIdentifier!
+            backgroundTaskID = UIApplication.shared.beginBackgroundTask(withName: "Keep Alive Timer") { [weak self] in
```

Comment:
I think a comment explaining what you just explained to me would be helpful: that we want a chance to expire our sessions without waiting for the app to be brought back into the foreground sometime in an arbitrarily distant future. If I have that right.

Reply:
You do, however I was explaining to you what the Apple docs arguably do a better job of. I.e. if you understand how Apple's background tasks work, then this code is very straightforward.

Reply:
I'm thinking more of how it interacts with our expiry algorithm.  eg.:

    // Best-effort attempt to expire our background sessions before app is killed in the background.